lib/scala/pkg/src/main/scala/org/enso/pkg/ComponentGroup.scala

```diff
@@ -0,0 +1,316 @@
+package org.enso.pkg
+
+import io.circe._
+import io.circe.syntax._
+
+/** The description of component groups provided by the package.
+  *
+  * @param newGroups the list of component groups provided by the package
+  * @param extendedGroups the list of component groups that this package extends
+  */
+case class ComponentGroups(
+  newGroups: List[ComponentGroup],
+  extendedGroups: List[ExtendedComponentGroup]
+)
+object ComponentGroups {
+
+  /** An empty component groups. */
+  val empty: ComponentGroups =
+    ComponentGroups(List(), List())
+
+  /** Fields for use when serializing the [[ComponentGroups]]. */
+  private object Fields {
+    val New     = "new"
+    val Extends = "extends"
+  }
+
+  /** Does the provided JSON object have unknown fields. */
+  private def hasUnknownKeys(cursor: HCursor): Boolean =
+    (ConfigCodecs.objectKeys(cursor) - Fields.New - Fields.Extends).nonEmpty
+
+  /** [[Encoder]] instance for the [[ComponentGroups]]. */
+  implicit val encoder: Encoder[ComponentGroups] = { componentGroups =>
+    val newGroups = Option.unless(componentGroups.newGroups.isEmpty)(
+      Fields.New -> componentGroups.newGroups.asJson
+    )
+    val extendsGroups = Option.unless(componentGroups.extendedGroups.isEmpty)(
+      Fields.Extends -> componentGroups.extendedGroups.asJson
+    )
+    Json.obj(newGroups.toSeq ++ extendsGroups.toSeq: _*)
+  }
+
+  /** [[Decoder]] instance for the [[ComponentGroups]]. */
+  implicit val decoder: Decoder[ComponentGroups] = { json =>
+    def decodeComponentGroups: Either[DecodingFailure, ComponentGroups] = for {
+      newGroups <- json.getOrElse[List[ComponentGroup]](Fields.New)(List())
+      extendsGroups <-
+        json.getOrElse[List[ExtendedComponentGroup]](Fields.Extends)(List())
+    } yield ComponentGroups(newGroups, extendsGroups)
+
+    if (hasUnknownKeys(json)) {
+      Left(
+        DecodingFailure(
+          "Invalid component groups. Valid keys are 'new' or 'extends'.",
+          json.history
+        )
+      )
+    } else {
+      decodeComponentGroups
+    }
+  }
+}
+
+/** The definition of a single component group.
+  *
+  * @param module the module name
+  * @param color the component group color
+  * @param icon the component group icon
```

Comment:
I'm not sure if this is the right place for this as it is probably documented somewhere in the IDE. But from this description I don't know what format the color is stored in - is it its English name or a hex RGB value like `#AABBCC`? Similarly for the icon - is it a filename? (But maybe this is not the place to document this, if so - please ignore this comment)

Reply:
I guess, for now, these values will be interpreted by IDE, and we'll just pass them as is